core/src/main/java/io/kestra/core/runners/Worker.java

```diff
@@ -628,13 +633,16 @@ public AtomicInteger getMetricRunningCount(WorkerTask workerTask) {
             ));
     }
 
+    /** {@inheritDoc} **/
+    @PreDestroy
     @Override
     public void close() throws Exception {
-        closeWorker(serverConfig.terminationGracePeriod());
+        if (shutdown.compareAndSet(false, true)) {
```

Comment:
Can you just confirm that this is to avoid double call to closeWorker?

Reply:
Yes, this is to avoid any issue if the close method is eventually invoked twice. I think it's always preferable to have some defensive code on this kind of method.